src/ui/components/results/verticalresultscomponent.js

```diff
@@ -254,9 +256,13 @@ export default class VerticalResultsComponent extends Component {
     };
   }
 
-  mount () {
+  mount (container) {
     if (Object.keys(this.getState()).length > 0) {
-      super.mount();
+      const searchState = this.getState('searchState');
+      Object.entries(SearchStates).forEach(([k, state]) =>
+        this.removeContainerClass(`yxt-Results--${snakeCaseToCamelCase(state)}`));
```

Comment:
would it be clearer to just do a switch here? This is kind of using the implementation detail that we use snake case strings as enums, in order to create a specific css class. It's also slightly harder to search for where `yxt-Results--{state}` classes come from

Reply:
I see you're point. Updating!